tests/test_delineateit.py

```diff
@@ -345,3 +349,112 @@ def test_check_geometries(self):
 
         target_layer = None
         target_vector = None
+
+    def test_detect_pour_points(self):
+        from natcap.invest.delineateit import delineateit
+ 
+        # create a flow direction raster from the sample DEM
+        flow_dir_path = os.path.join(REGRESSION_DATA, 'input/flow_dir_gura.tif')
+        output_path = os.path.join(self.workspace_dir, 'point_vector.gpkg')
+
+        delineateit.detect_pour_points((flow_dir_path, 1), output_path)
+
+        vector = gdal.OpenEx(output_path, gdal.OF_VECTOR)
+        layer = vector.GetLayer()
+
+        points = []
+        for feature in layer:
+            geom = feature.GetGeometryRef()
+            x, y, _ = geom.GetPoint()
+            points.append((x, y))
+        points.sort()
+
+        expected_points = [
+            (277713.1562500205, 9941874.499999935),
+            (277713.1562500205, 9941859.499999935)]
+
+        self.assertTrue(numpy.isclose(points[0][0], expected_points[0][0]))
+        self.assertTrue(numpy.isclose(points[0][1], expected_points[0][1]))
+        self.assertTrue(numpy.isclose(points[1][0], expected_points[1][0]))
+        self.assertTrue(numpy.isclose(points[1][1], expected_points[1][1]))
+
+    def test_calculate_pour_point_array(self):
+        from natcap.invest.delineateit import delineateit, delineateit_core
+
+        a = 100  # nodata value
+
+        # at this point the flow direction array will already have been
+        # buffered with a border of nodata
+        flow_dir_array = numpy.array([
+            [7, 7, 7, 5],
+            [6, 6, 6, 4],
+            [0, 1, a, 0],
+            [a, 2, 3, 4]
+        ], dtype=numpy.intc)
+
+        edges = numpy.array([1, 1, 0, 0], dtype=numpy.intc)  # top, left, bottom, right
+
+        expected_pour_points = {(6.25, 5.25)}
+
+        output = delineateit_core.calculate_pour_point_array(
+            flow_dir_array, 
+            edges, 
+            nodata=a,
+            offset=(0, 0),
+            origin=(5, 3),
+            pixel_size=(0.5, 1.5))
+
+        self.assertTrue(numpy.array_equal(
+            output,
+            expected_pour_points))
+
+    def test_find_pour_points_by_block(self):
+        from natcap.invest.delineateit import delineateit
+
+        a = 100  # nodata value
+        flow_dir_array = numpy.array([
+            [0, 0, 0, 0, 7, 7, 7, 1, 6, 6],
+            [2, 3, 4, 5, 6, 7, 0, 1, 1, 2],
+            [2, 2, 2, 2, 0, a, a, 3, 3, a],
+            [2, 1, 1, 1, 2, 6, 4, 1, a, a],
+            [1, 1, 0, 0, 0, 0, a, a, a, a]
+        ], dtype=numpy.int8)
+
+        srs = osr.SpatialReference()
+        srs.ImportFromEPSG(3157)
+        projection_wkt = srs.ExportToWkt()
+        gtiff_creation_tuple_options = ('GTIFF', (
+            'TILED=YES', 'BIGTIFF=YES', 'COMPRESS=LZW',
+            'BLOCKXSIZE=4', 'BLOCKYSIZE=5'))

```

Comment:
Block sizes typically need to be powers of 2 ... does `BLOCKYSIZE=5` do what we'd expect?

Reply:
No longer need this! Seems to be impractical to create the raster with these small block sizes, so I'm going with the `mock.patch` approach